agent/eventhandler/task_handler.go

```diff
@@ -184,14 +184,16 @@ func (handler *TaskHandler) taskStateChangesToSend() []api.TaskStateChange {
 		// safety mechanism) and add it to the list of task state changes
 		// that need to be sent to ECS
 		if task, ok := handler.state.TaskByArn(taskARN); ok {
-			event := api.TaskStateChange{
-				TaskARN: taskARN,
-				Status:  task.GetKnownStatus(),
-				Task:    task,
+			if task.GetKnownStatus() < api.TaskStopped {
```

Comment:
1. There's a remote possibility that task's known status would change between lines 187 and 190. Can you assign it to a variable instead of getting it twice? 
2. Also, you can reduce the levels of indentation by doing something like:
```
taskKnownStatus := task.GetKnownStatus()
if taskKnownStatus >= api.TaskStopped {
 continue
}
event := api.TaskStateChange {
 ...
}
...
```

Reply:
sounds good to me. fixed.